contributing/development/core_and_modules/internal_audio_architecture.rst

```diff
@@ -0,0 +1,236 @@
+.. _doc_internal_audio_architecture:
+
+Internal audio architecture
+===========================
+
+This page is a high-level overview of Godot 4's internal audio design.
+It does not apply to previous Godot versions.
+
+The goal of this page is to document design decisions taken to best suit
+:ref:`Godot's design philosophy <doc_best_practices_for_engine_contributors>`,
+while providing a starting point for new audio contributors.
+
+If you have questions about audio internals not answered here, feel free to
+ask in the ``#audio`` channel of the
+`Godot Contributors Chat <https://chat.godotengine.org/channel/audio>`__.
+
+
+.. note::
+
+    If you have difficulty understanding concepts on this page,
+    it is recommended to go through an audio blog such as
+    `Audio Sample Rate and Bit Depth <https://www.izotope.com/en/learn/digital-audio-basics-sample-rate-and-bit-depth.html>`__,
+    `How Sound Works <https://audiouniversityonline.com/audio-basics-how-sound-works/>`__, and
+    `An Introduction to Audio Electronics <https://www.allaboutcircuits.com/technical-articles/introduction-audio-electronics-sound-microphones-speakers-amplifiers>`__.
+
+
+.. _doc_internal_audio_architecture_drivers:
+
+Audio Drivers
+-------------
+
+Supported
+- Dummy
+- linuxBSD
+
+   - Alsa
+   - PulseAudio
+   - Alsa MIDI
+
+- Windows + UWP
+
+   - WASAPI
+   - XAudio2
+   - Win MIDI
+
+- macOS + iOS
+
+   - CoreAudio
+   - Core MIDI
+
+- Android
+
+   - OpenSL ES
+
+
+.. _doc_internal_audio_architecture_codecs:
+
+
+Audio Codecs
+------------
+
+
+OGG Vorbis
+^^^^^^^^^^
+
+.. note::
+
+    This is an audio format that takes a bit longer than WAV to play,
+    but is smaller in file size than WAV and MP3. Therefore most suitable for playing music in-game.
+
+
+**Why not vorbisfile?**
+// TODO
+
+
+WAV
+^^^
+
+.. note::
+
+    This is an audio format that is the qickest to load, but is larger in file size than Ogg Vorbis and MP3. Therefore most suitable for playing sound effects.
+
+
+MP3
+^^^
+
+
+.. _doc_internal_audio_architecture_servers:
+
+AudioServer
+------------
```

Comment:
Title underline must match the title's length (use a fixed-width font in your editor to notice this more easily):

```suggestion
AudioServer
-----------
```

Reply:
Thanks for catching that.  
I do use a monospaced font, but I copy&paste the underlines because the underline typing is annoying.